src/System.Management.Automation/security/SecureStringHelper.cs

```diff
@@ -346,44 +340,33 @@ internal static SecureString Decrypt(string input, byte[] key, byte[] IV)
             Utils.CheckArgForNullOrEmpty(input, "input");
             Utils.CheckKeyArg(key, "key");
 
-            byte[] decryptedData = null;
-            byte[] encryptedData = null;
-            SecureString s = null;
-
             //
             // prepare the crypto stuff
             //
-            Aes aes = Aes.Create();
-            encryptedData = ByteArrayFromString(input);
-
-            var decryptor = aes.CreateDecryptor(key, IV ?? aes.IV);
-
-            MemoryStream ms = new MemoryStream(encryptedData);
-
-            using (CryptoStream cs = new CryptoStream(ms, decryptor, CryptoStreamMode.Read))
+            using (var aes = Aes.Create())
             {
-                byte[] tempDecryptedData = new byte[encryptedData.Length];
-
-                int numBytesRead = 0;
-
-                //
-                // decrypt the data
-                //
-                numBytesRead = cs.Read(tempDecryptedData, 0,
-                                       tempDecryptedData.Length);
-
-                decryptedData = new byte[numBytesRead];
-
-                for (int i = 0; i < numBytesRead; i++)
+                using (var decryptor = aes.CreateDecryptor(key, IV ?? aes.IV))
```

Comment:
```suggestion
                using (ICryptoTransform decryptor = aes.CreateDecryptor(key, IV ?? aes.IV))
```

Reply:
We seem to use the `IV ?? aes.IV` pattern here and a separate assignment above. Might be worth changing the one above while we're improving the code?

Reply:
The separate assignment above is needed because the initialization vector is also returned with the encrypted data.